### PR TITLE
Fix retry mechanism to support error 500 and socket timeouts

### PR DESCRIPTION
**Pull request by sbenita**
Signed-off-by: Shlomi Benita <shlomib@cyberark.com>

### What does this PR do?
This PR fixes the retry mechanism of error 500 and socket timeouts.
The previous mechanism didn't catch these errors since they are raising an exception.
I added one more tests to verify that in case of other error like certificate problem it doesn't trigger the mechanism.
I did manual testing to verify that the mechanism works in case of timeouts and Error 500.

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

**Comment by rpothier** [comment]
Thanks for fixing this!

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/Users/Shlomi.Benita/.pyenv/shims/python"
```
Comment (rpothier) [inline .vscode/settings.json:2]: This points to a local env setting, can this be removed?

Comment (sbenita) [reply .vscode/settings.json]: Removed. sorry.

**Comment by rpothier** [comment]
Looks good! Were you verify that these new tests are running? When the tests pass I don't see any output in the logs.

**Comment by sbenita** [comment]
I added another test to verify that in case of invalid certificate we won't enter to the retry mechanism since it's not recoverable error.

The rest of the tests were done manually to verify that in each case of 500 or socket timeout it goes to the retry mechanism

**Review by rpothier — APPROVED** [review]

**Pull request merged**